server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1593,4 +1604,16 @@ public void setDefaultSearchPipeline(String defaultSearchPipeline) {
             throw new SettingsException("Unsupported setting: " + DEFAULT_SEARCH_PIPELINE.getKey());
         }
     }
+
+    public void setIndexConcurrentSegmentSearchEnabled(boolean indexConcurrentSegmentSearchEnabled) {
+        if (FeatureFlags.isEnabled(CONCURRENT_SEGMENT_SEARCH)) {
+            this.indexConcurrentSegmentSearchEnabled = indexConcurrentSegmentSearchEnabled;
+        } else {
+            throw new SettingsException("Unsupported setting: " + INDEX_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey());
```

Comment:
For this I was following the same convention as `SEARCH_PIPELINE`, which looks like it is also currently gated by a feature flag.

https://github.com/opensearch-project/OpenSearch/blob/a412d570049bfcd31dc6154c29e821e00e93e1b3/server/src/main/java/org/opensearch/index/IndexSettings.java#L1600-L1606

I think we should keep it consistent and since the search pipeline one is already released as so I think we should keep the concurrent search setting the same.

Reply:
Feature flag is fine - we should provide the explanation to the user why this setting is not accepted

Reply:
Right what I meant is that the other feature flag gated index setting has the same exception with no details so I was trying to be consistent with that.

Reply:
We should change the other places instead